include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -417,6 +417,53 @@ def MulFOpIEEE754 : ArithBinaryFloatingPointLibraryOp<"ieee754.mul", [
   }];
 }
 
+// This models the division and square root, distinguished by `sqrtOp`,
+// operation interface in Berkeley HardFloat. It computes `left`/`right`
+// if `sqrtOp` = 0; it ignores `right` and computes sqrt(`left`) if `sqrtOp` = 1.
+def DivSqrtOpIEEE754 : ArithBinaryFloatingPointLibraryOp<"ieee754.divSqrt", [
+    SameTypeConstraint<"left", "out">
+  ]> {
+  let results = (outs I1:$clk, I1:$reset, I1:$go, I1:$control, I1:$sqrtOp,
+    AnySignlessInteger:$left, AnySignlessInteger:$right, AnySignlessInteger:$roundingMode, AnySignlessInteger:$out,
```

Comment:
As discussed offline, let's use the correct `I<N>` type for the values that are not parametric over width.

Reply:
Done refactoring!